test/releasers/dotnet.ts

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {describe, it, afterEach} from 'mocha';
+import * as nock from 'nock';
+import {DotNet} from '../../src/releasers/dotnet';
+import {readFileSync} from 'fs';
+import {resolve} from 'path';
+import * as snapshot from 'snap-shot-it';
+import * as suggester from 'code-suggester';
+import * as sinon from 'sinon';
+
+const sandbox = sinon.createSandbox();
+const fixturesPath = './test/releasers/fixtures/dotnet';
+
+function getBlob(filename: string) {
+  const content = readFileSync(resolve(fixturesPath, filename), 'utf8').replace(
+    /\r\n/g,
+    '\n'
+  );
+  return {
+    content: Buffer.from(content, 'utf8').toString('base64'),
+    sha: 'abc123',
+  };
+}
+
+describe('DotNet', () => {
+  afterEach(() => {
+    sandbox.restore();
```

Comment:
When using `nock`, I like to do a `nock.cleanAll()` in the after to catch any floating nocked enpoints between tests.

Reply:
I call `nock.done()` as an assertion for all tests, which will throw an exception if any mocks haven't been satisfied.

Reply:
Sure, and to be clear, this isn't a big deal.  There are two places where this matters:
- sometimes you do forget to do a `req.done()` :) I've made that mistake many times, and it's hard to catch.
- If you *do* have a test that starts to throw or fail, nock will keep the nocked endpoints around and cause havoc with tests further down the chain. 

It's safer to just clear it, and be sure the failures you see are for the specific test that fails.  But again - not a huge deal.